### DISCUSSION TITLE
MapView in all Mapsui Targets.

### DISCUSSION THREAD

**Discussion by inforithmics** [Mapsui/Mapsui#1431] [Ideas]
Currently MapView ist only available for Maui and Forms. 
But I think the other Targets could use it too. And it would unify Mapsui with the Same Controls everywhere. I had a quick Look at MapView and it could be possible to use the Same technic for sharing Like the MapControl. So my question is, is there Interest for this or opinions or roadblocks. Because I' m not an expert in this Area.

**Comment by charlenni** [comment]
What part of MapView do you want to have on all platforms? The WGS84 coordinate system? The easy to use pin and drawables? Or anything else?

**Reply by inforithmics** [reply]
In the End I would have the part of MapView that is Platform independent on all platforms. 
So I guess the WGS84 Coordinate System and pins and drawables would be a first step in that direction.
But I don't know if there is broader interest for that. 
But I guess that when someone wants to display only a Map the MapControl is pretty good for that. But for interaction like setting Pins and things like that The MapView is better suited.
But this is only my personal opinion. 
I wanted to start a discussion here what the "general" opinion - sentiment is about this idea?

**Comment by charlenni** [comment]
The idea behind MapView was to have a replacement for the normal Xamarin.Forms Map library or better, for a Google free library. MapView works much more like normal Map libraries than Mapsui by itself, so it could be a good thing to have a convenience layer above Mapsui.

**Comment by pauldendulk** [comment]
I am a big proponent of having as much code as possible platform independent. Even for MapControl there can be a lot of improvements. Like click logic. Handling widget click events. BoundingBox drag. 

For MapView there is much work to do to move that way. We should also think about the distinction between MapControl and MapView. There do not have to be that many differences. Perhaps part of the code in MapView should be in MapControl and perhaps MapView could be a small extensions. Perhaps even an optional thing on MapControl. 

One of the main things is that conversion to and from WGS84 (GPS coordinates). This works because MapView assumes a SphericalMercator projection. MapControl theoretically allows other projections. If the projection is not supported, some things just don't work. We should handle this gracefully, but we do not have to add an all different map control for that.
